common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -39,8 +37,16 @@ public static void cleanDirectBuffer(ByteBuffer buffer) {
       return;
     }
     if (buffer.isDirect()) {
-      Cleaner cleaner = ((DirectBuffer) buffer).cleaner();
-      cleaner.clean();
+      try {
+        final Method getCleanerMethod = buffer.getClass().getMethod("cleaner");
+        getCleanerMethod.setAccessible(true);
+        final Object cleaner = getCleanerMethod.invoke(buffer);
+        if (cleaner != null) {
+          cleaner.getClass().getMethod("clean").invoke(cleaner);
+        }
+      } catch (Exception e) {
```

Comment:
Why not throw a wrapped exception showing that it fails to unmap the buffer and telling the cause? What's the reasoning behind `buffer = null` ?